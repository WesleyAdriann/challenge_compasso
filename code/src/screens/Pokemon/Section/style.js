import styled from 'styled-components/native'

import { COLORS, globalStyles } from '../../../style'

export const Container = styled.View`
  padding: 8px 8px 16px;
  border-bottom-width: 1px;
  border-color: ${COLORS.primaryRed};
`

export const Title = styled.Text`
  ${globalStyles.defaultFont}
  font-size: 16px;
  font-weight: bold;
  margin-bottom: 6px;
`

export const Text = styled.Text`
  ${globalStyles.defaultFont}
`
