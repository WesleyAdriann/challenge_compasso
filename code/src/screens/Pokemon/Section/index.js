import React from 'react'
import PropTypes from 'prop-types'

import { Container, Title, Text } from './style'

const Section = ({ title, children, texts }) => (
  <Container>
    <Title>{title}</Title>
    {
      texts.length
        ? texts.map((text, index) => (
        <Text key={index}>{text}</Text>
          ))
        : children
    }
  </Container>
)

Section.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node,
  texts: PropTypes.arrayOf(PropTypes.string)
}

Section.defaultProps = {
  texts: []
}

export default Section
