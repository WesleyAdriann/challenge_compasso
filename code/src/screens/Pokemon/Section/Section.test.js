import React from 'react'
import renderer from 'react-test-renderer'

import Section from './index'

describe('Section Component', () => {
  test('renders correctly', () => {
    const props = {
      title: 'renders correctly test'
    }

    const tree = renderer.create(<Section {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
    
  })
})
