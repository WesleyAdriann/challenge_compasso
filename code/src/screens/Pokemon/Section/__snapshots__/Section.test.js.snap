// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#D50000",
        "paddingBottom": 16,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#050505",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 6,
        },
      ]
    }
  >
    renders correctly test
  </Text>
</View>
`;
